torch/distributed/algorithms/ddp_comm_hooks/mixed_precision_hooks.py

```diff
@@ -38,7 +39,8 @@ def _reducer_allreduce_and_upcast_hook(
     fut = reducer._run_allreduce_hook(bucket)
     ret_fut = torch.futures.Future()
     stream = hook_state.upcast_stream
-    with torch.cuda.stream(stream):
+    device_module = _get_device_module(stream.device.type)
```

Comment:
This is a hot path that runs during gradient allreduce, so we should cache device_module look up when the hoook state is initialized.